### PR TITLE
Update StartsWith invocations to use appropriate StringComparison

### PR DESCRIPTION
**Pull request by VladimirReshetnikov**
Partial fix for #409
Update StartsWith invocations to use appropriate StringComparison (part 1)

### CODE REVIEW

**Comment by tmeschter** [comment]
Do we need an analyzer to enforce that you must pass a `StringComparison` to `StartsWith`?

**Comment by VladimirReshetnikov** [comment]
@tmeschter Yes, I think so.

**Comment by jasonmalinowski** [comment]
:+1:

File: src/Features/CSharp/Completion/CompletionProviders/SnippetCompletionProvider.cs [hunk 1]
```diff
@@ -134,7 +134,7 @@ public override bool SendEnterThroughToEditor(CompletionItem completionItem, str
             var snippets = service.GetSnippetsIfAvailable();
             if (isPreProcessorContext)
             {
-                snippets = snippets.Where(snippet => snippet.Shortcut.StartsWith("#"));
+                snippets = snippets.Where(snippet => snippet.Shortcut.StartsWith("#", StringComparison.Ordinal));
```
Comment (pharring) [inline src/Features/CSharp/Completion/CompletionProviders/SnippetCompletionProvider.cs]: snippet.Shortcut[0] == '#' ?

File: src/VisualStudio/Core/Def/Implementation/GenerateType/GenerateTypeDialogViewModel.cs [hunk 2]
```diff
@@ -257,7 +257,7 @@ internal bool TrySubmit()
                     return false;
                 }
 
-                var isRootOfTheProject = trimmedFileName.StartsWith("\\");
+                var isRootOfTheProject = trimmedFileName.StartsWith(@"\", StringComparison.Ordinal);
```
Comment (pharring) [inline src/VisualStudio/Core/Def/Implementation/GenerateType/GenerateTypeDialogViewModel.cs]: trimmedFileName[0] == @'\';
?

File: src/VisualStudio/Core/Def/Implementation/GenerateType/GenerateTypeDialogViewModel.cs [hunk 3]
```diff
@@ -277,10 +277,10 @@ internal bool TrySubmit()
                 }
 
                 // Remove the '\' at the beginning if present
-                trimmedFileName = trimmedFileName.StartsWith("\\") ? trimmedFileName.Substring(1) : trimmedFileName;
+                trimmedFileName = trimmedFileName.StartsWith(@"\", StringComparison.Ordinal) ? trimmedFileName.Substring(1) : trimmedFileName;
```
Comment (pharring) [inline src/VisualStudio/Core/Def/Implementation/GenerateType/GenerateTypeDialogViewModel.cs]: trimmedFileName = isRootOfTheProject ? trimmedFileName.Substring(1) : trimmedFileName;

File: src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs [hunk 4]
```diff
@@ -230,7 +230,7 @@ private static bool RemovalMayIntroduceCastAmbiguity(ParenthesizedExpressionSynt
                 if (expression.IsKind(SyntaxKind.NumericLiteralExpression))
                 {
                     var numericLiteral = (LiteralExpressionSyntax)expression;
-                    if (numericLiteral.Token.ValueText.StartsWith("-"))
+                    if (numericLiteral.Token.ValueText.StartsWith("-", StringComparison.Ordinal))
```
Comment (pharring) [inline src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs]: numericLiteral.Token.ValueText[0] == '-'

Comment (pharring) [inline src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs]: ... oh, I guess you have to worry about empty strings.

File: src/Workspaces/CSharp/Portable/Extensions/StringExtensions.cs [hunk 5]
```diff
@@ -39,14 +40,14 @@ internal static class StringExtensions
                 return SyntaxFactory.Identifier(escaped);
             }
 
-            var unescaped = identifier.StartsWith("@")
+            var unescaped = identifier.StartsWith("@", StringComparison.Ordinal)
```
Comment (pharring) [inline src/Workspaces/CSharp/Portable/Extensions/StringExtensions.cs]: Looks like ```identifier.StartsWith("@", StringComparison.Ordinal)``` could be extracted into a local bool (used again on line 50)

**Pull request merged**